addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials.py

```diff
@@ -175,15 +175,28 @@ def __gather_orm_texture(blender_material, export_settings):
 
     metallic_socket = gltf2_blender_get.get_socket_or_texture_slot(blender_material, "Metallic")
     roughness_socket = gltf2_blender_get.get_socket_or_texture_slot(blender_material, "Roughness")
-    if metallic_socket is None or roughness_socket is None\
-            or not __has_image_node_from_socket(metallic_socket)\
-            or not __has_image_node_from_socket(roughness_socket):
+
+    hasMetal = metallic_socket is not None and __has_image_node_from_socket(metallic_socket)
+    hasRough = roughness_socket is not None and __has_image_node_from_socket(roughness_socket)
+
+    if not hasMetal and not hasRough:
         metallic_roughness = gltf2_blender_get.get_socket_or_texture_slot_old(blender_material, "MetallicRoughness")
         if metallic_roughness is None or not __has_image_node_from_socket(metallic_roughness):
             return None
-        return (occlusion, metallic_roughness, metallic_roughness)
+        result = (occlusion, metallic_roughness)
+    elif not hasMetal:
+        result = (occlusion, roughness_socket)
+    elif not hasRough:
+        result = (occlusion, metallic_socket)
+    else:
+        result = (occlusion, roughness_socket, metallic_socket)
+
+    # Double-check this will past the filter in texture_info (otherwise there are different resolutions or other problems).
+    info = gltf2_blender_gather_texture_info.gather_texture_info(result, export_settings)
+    if info is None:
+        return None
 
-    return (occlusion, roughness_socket, metallic_socket)
+    return result
```

Comment:
This method could return either `(o, r)` or `(o, m)` among other options. I can't see where/how we later resolve whether that second entry should be written to the blue or green channel. Should we be returning `(o, None, m)` or something that matches up with the channels like that?

Reply:
This is due to architecture that was already in place before my changes.  The channel assignments are (re-) computed in the [gather_image](https://github.com/KhronosGroup/glTF-Blender-IO/blob/d2405aedb0b5ecd5f65fd6cb851131e3d772e017/addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_image.py#L166) helper.  If `None` shows up anywhere in this result tuple, the entire image will be filtered out (by several different filter routines, sprinkled around).

With current architecture, `None` is forbidden here, and order is irrelevant.  The channel assignments come from inspecting the node socket connections.

Reply:
This is why, in `master` currently, if either roughness or metallic is left disconnected, the entire RM map will be missing.  It gets filtered out.

Reply:
Ok, thank you! This code looks good to me, then.